bin/gff_mapping.py

```diff
@@ -30,46 +32,49 @@ def mobilome_parser(mobilome_prokka_extra):
-                l_line = line.rstrip().split("\t")
-                # Annotation lines have exactly 9 columns
-                if len(l_line) == 9:
-                    contig = l_line[0]
-                    annot_source = l_line[1]
-                    if annot_source in source_tools:
-                        if contig in mobilome_annot:
-                            mobilome_annot[contig].append(line.rstrip())
-                        else:
-                            mobilome_annot[contig] = [line.rstrip()]
+    with open(mobilome_clean, "r") as input_table:
+        for line in input_table:
+            l_line = line.rstrip().split("\t")
+            # Annotation lines have exactly 9 columns
+            if len(l_line) == 9:
+                contig = l_line[0]
+                annot_source = l_line[1]
+                seq_type = l_line[2]
+                start = int(l_line[3])
+                end = int(l_line[4])
+                strand = l_line[6]
+                coordinates = (start, end)
+                if annot_source in source_tools:
+                    composite_key = ( contig, start, end )
+                    mob_types[composite_key] = seq_type                        
+                    if contig in mobilome_annot:
+                        mobilome_annot[contig].append(line.rstrip())
+                        mges_dict[contig].append(coordinates)
                     else:
-                        start = l_line[3]
-                        end = l_line[4]
-                        strand = l_line[6]
-                        composite_key = (contig, start, end, strand)
-                        attrib = l_line[8]
-                        extra_list = []
-                        for attr in attrib.split(";"):
-                            att_key = attr.split("=")[0]
-                            att_val = attr.split("=")[1]
-                            if att_key in extra_annot:
-                                extra_list.append(attr)
-                        if len(extra_list) > 0:
-                            extra_val = ";".join(extra_list)
-                            proteins_annot[composite_key] = extra_val
-
-    return (proteins_annot, mobilome_annot)
+                        mobilome_annot[contig] = [line.rstrip()]
+                        mges_dict[contig] = [coordinates]
+                else:
+                    str_composite_key = (contig, str(start), str(end), strand)
+                    attrib = l_line[8]
+                    extra_list = []
+                    for attr in attrib.split(";"):
+                        att_key = attr.split("=")[0]
+                        att_val = attr.split("=")[1]
+                        if att_key in extra_annot:
+                            extra_list.append(attr)
+                    if len(extra_list) > 0:
+                        extra_val = ";".join(extra_list)
+                        proteins_annot[str_composite_key] = extra_val
+                        
```

Comment:
extra whitespace

Reply:
ah, yes.. there is a bit of a mixture of tabs and spaces on this repo. I will tidy up the code with prettier